flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/adaptive/Canceling.java

```diff
@@ -55,18 +66,7 @@ public void cancel() {
     }
 
     @Override
-    public void handleGlobalFailure(Throwable cause) {
-        getLogger()
-                .debug(
-                        "Ignored global failure because we are already canceling the job {}.",
-                        getJobId(),
-                        cause);
-    }
-
-    @Override
-    boolean updateTaskExecutionState(TaskExecutionStateTransition taskExecutionStateTransition) {
-        return getExecutionGraph().updateState(taskExecutionStateTransition);
-    }
+    void onFailure(Throwable failure) {}
```

Comment:
```suggestion
    void onFailure(Throwable failure) {
      // Why do we ignore failures in this state?
    }
```

applies to multiple places